### PR TITLE
fix: remove deadlock possibility by adding resend_queue queue

### LINKED ISSUE #71: Agent does not send any data after throttling event
Hello,

We had a situation when output was completely blocked after hitting a throttling event on Sumo side. The data started to flow only after bouncing pipeline. The logs below are the last messages in the log. Nothing else was logged till the bounce. The issue happened in four different data centers at the same time (same target error).

[2022-01-31T06:45:44,647][ERROR][logstash.outputs.sumologic][sumo_out] request rejected {:token=>64, :code=>429, :headers=>{"X-Sumo-Client"=>"logstash-output-sumologic", "X-Sumo-Category"=>"ca/elastiflow", "X-Sumo-Host"=>"my.domain", "X-Sumo-Name"=>"elastiflow", "Content-Type"=>"text/plain"}, :contet=>"{\"event\":{\"type\":\"net"}
[2022-01-31T06:45:45,713][ERROR][logstash.outputs.sumologic][sumo_out] request rejected {:token=>6, :code=>429, :headers=>{"X-Sumo-Client"=>"logstash-output-sumologic", "X-Sumo-Category"=>"ca/elastiflow", "X-Sumo-Host"=>"my.domain", "X-Sumo-Name"=>"elastiflow", "Content-Type"=>"text/plain"}, :contet=>"{\"event\":{\"type\":\"net"}

Thank you.

### PR DESCRIPTION
**Pull request by sumo-drosiek**
In order to send data simultaneously, one of two approaches can be used here:
 - rewriting code to be explicitly multithred
 - Use async as it is done already
 
This PR keeps the async approach and prevent deadlock by adding additional queue (sized twice as worker numbers)

I wanted to add minimal set of changes in order to fix #71

### CODE REVIEW

File: lib/logstash/outputs/sumologic/sender.rb [hunk 1]
```diff
@@ -39,9 +43,24 @@ def start()
       @stopping.make_false()
       @sender_t = Thread.new {
         while @stopping.false?
-          batch = @queue.deq()
+          begin
+            # Resend batch if any in the queue
+            batch = @resend_queue.deq(non_block: true)
+          rescue
```
Comment (astencel-sumo) [inline lib/logstash/outputs/sumologic/sender.rb:49]:
```suggestion
          rescue ThreadError
```

**Comment by swiatekm-sumo** [comment]
Instead of doing this, couldn't we just make the requeue non-blocking and drop data if there's no space left in the queue? Do we know what the right thing to do in general is for a Logstash output if it can't send data fast enough? Should it block, return an error, or something else?

**Comment by sumo-drosiek** [comment]
AFAIK if queue is full it is going to backpressure logstash, so no new events are going to be put into queue until it has space

> Should it block, return an error, or something else?
not sure, so leaving main architecture as it is

I want to avoid changing architecture or behavior if there is no need to do that

**Comment by sumo-drosiek** [comment]
I'm merging this PR as we need bugfix urgently. We can continue discussion here and add changes as separate PRs

**Pull request merged**